### ISSUE TITLE
Kirby and the Amazing Mirror fail to communcate with minigames using Single Pak Mode

### ISSUE THREAD

**Issue by Testsr** [mgba-emu/mgba#3295]
Reproduction steps

1. Boot one GBA with the game, and the other with BIOS.
2. Select a minigame, Multiplayer and when the other GBA is detected, click release.

What happens:
The minigame transfers properly, but the Level selector never appears, and the games lose communication after a second.

This bug is tested on Old branch, and New-sio branch.

**Comment by Testsr** [comment]
Will have bug report txt in a day or two.

**Comment by Testsr** [comment]
Also as a reminder, the four player bug on KaTaM needs to be addressed.

**Comment by endrift** [comment]
Seeing a lot of "GBA Serial I/O: Transfer restarted unexpectedly" in the logs. I haven't investigated further yet though.

**Comment by Testsr** [comment]
@endrift  Found decomp of Kirby and the Amazing Mirror at https://github.com/jiangzhengwenjz/katam, with SIO routines already decomplied and commented.

**Comment by endrift** [comment]
Yeah I saw that too. That's only the sending side that's commented. It'd be good to have the receiving side commented too, but we'll see what I can manage.
